dfiq.py

```diff
@@ -0,0 +1,523 @@
+        those components. Because of this, DFIQ takes a "bottom-up" approach to construct the hierarchy
+        at the time of initialization (using a networkx DiGraph).
+        """
+        if not self.graph:
+            raise ValueError("DFIQ Graph needed before adding children.")
+
+        for dfiq_id, component in self.components.items():
+            children = sorted(list(nx.DiGraph.successors(self.graph, dfiq_id)))
+            self.components[dfiq_id].set_children(children)
+
+    def add_child_tags(self) -> None:
+        """Adds tags from a Question's Approaches to that Question."""
+        if not self.graph:
+            raise ValueError("DFIQ Graph needed before adding children.")
+
+        for dfiq_id, component in self.components.items():
+            parents = sorted(list(nx.DiGraph.predecessors(self.graph, dfiq_id)))
+            if self.components[dfiq_id].type == "approach":
+                for parent in parents:
+                    self.components[parent].all_tags = self.components[
+                        parent
+                    ].all_tags.union(self.components[dfiq_id].tags)
+
+    @staticmethod
+    def convert_yaml_object_to_dfiq_component(
+        yaml_object,
+    ) -> Scenario | Facet | Question | Approach | None:
+        """Takes a dict, loaded from a DFIQ YAML file, and converts to the appropriate dfiq.Component object."""
+
+        assert yaml_object["type"] in (
+            "scenario",
+            "facet",
+            "question",
+            "approach",
+        ), "Object must be of known DFIQ type"
+
+        if yaml_object["type"] == "scenario":
+            return Scenario(
+                yaml_object["id"],
+                yaml_object["display_name"],
+                yaml_object.get("description"),
+                yaml_object.get("tags"),
+            )
+
+        elif yaml_object["type"] == "facet":
+            return Facet(
+                yaml_object["id"],
+                yaml_object["display_name"],
+                yaml_object.get("description"),
+                yaml_object.get("tags"),
+                yaml_object.get("parent_ids"),
+            )
+
+        elif yaml_object["type"] == "question":
+            return Question(
+                yaml_object["id"],
+                yaml_object["display_name"],
+                yaml_object.get("description"),
+                yaml_object.get("tags"),
+                yaml_object.get("parent_ids"),
+            )
+
+        elif yaml_object["type"] == "approach":
+            return Approach(
+                yaml_object["id"],
+                yaml_object["display_name"],
+                yaml_object.get("description"),
+                yaml_object.get("tags"),
+                yaml_object.get("view"),
+            )
+
+        else:
+            return None
+
+    def load_yaml_files_by_type(self, dfiq_type, yaml_data_path=None) -> dict:
```

Comment:
I notice that you use type hints inconsistently. I would opt to add it to all methods (in a followup PR is fine)

Reply:
I added more (hopefully all?) in